src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java

```diff
@@ -458,6 +435,26 @@ private void insert(final SinkRecord record)
 
     }
 
+    private boolean shouldFlush()
+    {
+      return (System.currentTimeMillis() - this.previousFlushTimeStamp) >= (getFlushTime() * 1000);
```

Comment:
only check time here. I didn't move the buffer size and record number triggered flush to here because this function will be called frequently. And only insert can trigger buffer size and record number based flush.